bentoml/_internal/bento/gen.py

```diff
@@ -116,6 +121,48 @@ def get_templates_variables(
     }
 
 
+def process_extend_blocks(templates: nodes.Template) -> None | list[str]:
+    """
+    Process the first block of the custom templates.
+    """
+    # If any extends block use bentoml templates,
+    # template value should be nodes.Const
+    template_list = [
+        v.value
+        for ext_block in templates.body
+        for _, v in ext_block.iter_fields()
+        if isinstance(ext_block, nodes.Extends)
+        and isinstance(ext_block.template, nodes.Const)
+    ]
+    return template_list if template_list else None
+
+
+def check_custom_template(tmpl: Template, *, use_conda: bool, use_cuda: bool) -> None:
+    if tmpl.filename is None and not isinstance(tmpl.filename, str):
```

Comment:
best effort check for all custom templates. We want to loosely do this to catch the above error.